awx/main/managers.py

```diff
@@ -127,14 +129,23 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
                         other_inst.save(update_fields=['ip_address'])
                         logger.warning("IP address {0} conflict detected, ip address unset for host {1}.".format(ip_address, other_hostname))
 
-            # Return existing instance that matches hostname
-            instance = self.filter(hostname=hostname)
+            # Return existing instance that matches hostname or UUID (default to UUID)
+            if uuid is not None and uuid != UUID_DEFAULT and self.filter(uuid=uuid).exists():
+                instance = self.filter(uuid=uuid)
```

Comment:
Before I approve, I want to echo a concern for which discussions happened in other channels.

My reading of the installer before this point was that they would generate a new uuid every run (different from real uuid if an existing host exists). I would like to be sure that doesn't happen for re-runs of the installer, otherwise I get nervous. What makes the _most_ sense to me is for the installer to read the `ha.py` file's SYSTEM_UUID value if it's an existing instance, and generate one only if it's a new instance.

At this moment in code, if `self.filter(hostname=hostname).exists()`, then it will result in an error. I'm fine with that because it should throw an error in that case.

Reply:
@AlanCoding yes, that makes sense. 

As you mentioned, the installer does generate a new `system_uuid` on every run, but it will only be used if the `ha.py` does not exist, **however**, at the task https://github.com/ansible/automation-platform-collection/blob/devel/roles/receptor/tasks/main.yml#L156-L159, we consume the `system_uuid` which is where the problem could happen. 

We will modify the installer to set the `system_uuid` from the `ha.py` and when `ha.py` then generate it fresh.

Reply:
Yeah, if reading from `ha.py` isn't needed for the current fix, and is just a matter of code cleanup, then feel free to file and issue to rework it later and get the currently needed fixes in. I think we're fully on the same page now.

Reply:
@rebeccahhh++  opened the PR https://github.com/ansible/automation-platform-collection/pull/179 which should take of this